Assets/MixedRealityToolkit.Services/DiagnosticsSystem/MixedRealityToolkitVisualProfiler.cs

```diff
@@ -152,6 +187,22 @@ private void Reset()
                 textMaterial = new Material(meshRenderer.sharedMaterial);
                 textMaterial.renderQueue = defaultMaterial.renderQueue;
                 Destroy(meshRenderer.gameObject);
+
+                MeshFilter quadMeshFilter = GameObject.CreatePrimitive(PrimitiveType.Quad).GetComponent<MeshFilter>();
+
+                if (defaultInstancedMaterial != null)
+                {
+                    // Create a quad mesh with artificially large bounds to disable culling for instanced rendering.
+                    // TODO: Use shared mesh with normal bounds once Unity allows for more control over instance culling.
+                    quadMesh = quadMeshFilter.mesh;
+                    quadMesh.bounds = new Bounds(Vector3.zero, Vector3.one * float.MaxValue);
+                }
```

Comment:
I am a bit confused. The [documentation](https://docs.unity3d.com/ScriptReference/Graphics.DrawMeshInstanced.html) for `Graphics.DrawMeshInstanced` says:
> Meshes are **not further culled** by the view frustum or baked occluders, nor sorted for transparency or z efficiency.

But there are [many](https://forum.unity.com/threads/instancing-and-culling-make-it-stop.511374/) [posts](https://forum.unity.com/threads/graphics-drawmeshinstanced-and-culling-the-how-and-why.633508/) about culling being performed in practice. So this sounds like a bug. Is there a Unity but report for this? I couldn't find any.

Note that for `DrawMeshInstancedIndirect` there is also some culling apparently, which you can work around via the `Bounds` parameter as described [here](https://answers.unity.com/questions/1349465/how-to-avoid-culling-when-instancing-object-via-dr.html) without having to change the mesh.

Reply:
This got me for a bit too, since I couldn't figure out why instances weren't being rendered. Once I looked near the origin they appeared, so I was able to narrow it down to a culling issue. Then I found the numerous Unity forum posts about the issue you pointed out. I'm not aware of a bug report, but it seems like there certainly is a bug, or their documentation is out of date. I have an open discussion with Unity around a similar issue so I'll reach out to them about this too!

Reply:
Thanks!